src/alarmhandler_p.h

```diff
@@ -0,0 +1,99 @@
+/*
+  This file is part of the mkcal library.
+
+  Copyright (c) 2023 Damien Caliste <dcaliste@free.fr>
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Library General Public
+  License as published by the Free Software Foundation; either
+  version 2 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Library General Public License for more details.
+
+  You should have received a copy of the GNU Library General Public License
+  along with this library; see the file COPYING.LIB.  If not, write to
+  the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+  Boston, MA 02110-1301, USA.
+*/
+/**
+  @file
+  This file is part of the API for handling alarms.
+
+  @author Damien Caliste \<dcaliste@free.fr\>
+*/
+
+#ifndef MKCAL_ALARMHANDLER_H
+#define MKCAL_ALARMHANDLER_H
+
+#include <KCalendarCore/Incidence>
+
+#include "mkcal_export.h"
+
+namespace mKCal {
+
+/**
+  @brief
+  This class provides an interface to handle alarms.
+*/
+class MKCAL_EXPORT AlarmHandler
```

Comment:
So this version now works with mkcaltool not using AlarmHandler directly, right? If so, do we need the export?

Btw, the emitStorage...() which got renamed recently is still having a bit confusing name as one could assume it mostly just emitting signals. Something like handleStorageUpdated() type of names could have been a bit better maybe.

Reply:
Indeed, I've removed it for the moment. I'll readd it when I'll drop extendedstorage from mkcaltool .